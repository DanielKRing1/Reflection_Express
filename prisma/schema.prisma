// This is your Prisma schema file,
// learn more about it in the docs: https://pris.ly/d/prisma-schema

generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL_LOCAL")
}

model User {
  password_userId             Password?           @relation(name: "password_userId")
  journal_userId              Journal[]           @relation(name: "journal_userId")
  email                       String              @id @unique

  // Set to null when delete Journal + Update id if Journal id updated
  user_lastUsedJournalId      Journal?            @relation(name: "user_lastUsedJournalId", fields: [lastUsedJId], references: [id], onDelete: SetNull, onUpdate: Cascade)
  lastUsedJId                 BigInt?             @unique
}

model Password {
  // Cascade delete/update with User
  password_userId             User                @relation(name: "password_userId", fields: [userId], references: [email], onDelete: Cascade, onUpdate: Cascade)
  userId                      String              @id

  hash                        String
}

model Journal {
  user_lastUsedJournalId      User?               @relation(name: "user_lastUsedJournalId")
  inkling_journalId           Inkling[]           @relation(name: "inkling_journalId")
  journalentry_journalId      JournalEntry[]      @relation(name: "journalentry_journalId")
  thought_journalId           Thought[]           @relation(name: "thought_journalId")
  reflection_journalId        Reflection[]        @relation(name: "reflection_journalId")
  id                          BigInt              @id @default(autoincrement())
  
  // Cascade delete/update with User
  journal_userId              User                @relation(name: "journal_userId", fields: [userId], references: [email], onDelete: Cascade, onUpdate: Cascade)
  userId                      String

  @@index([userId])
}

model Inkling {
  timeId                      DateTime            @db.Timestamptz(3)

  // Cascade delete/update with Journal
  inkling_journalId           Journal             @relation(name: "inkling_journalId", fields: [journalId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  journalId                   BigInt

  text                        String

  @@id([journalId, timeId])
}

model JournalEntry {
  reflection_journalEntryId   Reflection[]        @relation(name: "reflection_journalEntryId")
  timeId                      DateTime            @db.Timestamptz(3)

  // Cascade delete/update with Journal
  journalentry_journalId      Journal             @relation(name: "journalentry_journalId", fields: [journalId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  journalId                   BigInt

  @@id([journalId, timeId])
}

model Thought {
  reflection_thoughtId        Reflection[]        @relation(name: "reflection_thoughtId")
  timeId                      DateTime            @db.Timestamptz(3)

  // Cascade delete/update with Journal
  thought_journalId           Journal             @relation(name: "thought_journalId", fields: [journalId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  journalId                   BigInt

  text                        String

  @@id([journalId, timeId])
}

model Reflection {
  // Cascade delete/update with Journal
  reflection_journalId        Journal             @relation(name: "reflection_journalId", fields: [journalId], references: [id], onDelete: Cascade, onUpdate: Cascade)
  journalId                   BigInt

  // Cascade delete/update with JournalEntry
  reflection_journalEntryId   JournalEntry        @relation(name: "reflection_journalEntryId", fields: [journalId, journalEntryId], references: [journalId, timeId], onDelete: Cascade, onUpdate: Cascade)
  journalEntryId              DateTime            @db.Timestamptz(3)

  // Cascade delete/update with Thought
  reflection_thoughtId        Thought             @relation(name: "reflection_thoughtId", fields: [journalId, thoughtId], references: [journalId, timeId], onDelete: Cascade, onUpdate: Cascade)
  thoughtId                   DateTime            @db.Timestamptz(3)

  decision                    Int                 @db.SmallInt

  @@id([journalEntryId, thoughtId])
}
