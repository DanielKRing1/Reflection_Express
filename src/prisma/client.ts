import { PrismaClient } from "./generated/client";

let prisma = new PrismaClient({
    // log: ["query"],
});

export default prisma;
