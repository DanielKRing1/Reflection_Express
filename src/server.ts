// EXPRESS
import express, { Express } from "express";
import cors from "cors";

// MIDDLEWARE
import loggingMiddleware from "./middlewares/logging.middleware";

// ROUTES
import genRouter from "./routes";

// CONFIG
import { PORT } from "./config/server.config";

// APP
const app: Express = express();

// MIDDLEWARE
app.use(cors());
// parse requests of content-type - application/json
app.use(express.json());
// parse requests of content-type - application/x-www-form-urlencoded
app.use(
  express.urlencoded({
    extended: true,
  })
);
app.use(loggingMiddleware);

// GRAPHQL
app.use(genRouter());

// START SERVER
app.listen(PORT, () => console.log(`Server listening on port ${PORT}!`));
