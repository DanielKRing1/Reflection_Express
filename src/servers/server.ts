import genExpressApp from "./app";
import gqlServer from "./gqlServer";

import accessPromise from "../middlewares/session/access";

export default (async () => {
    const app = await genExpressApp();
    const access = await accessPromise();

    return gqlServer(app, access);
})();
