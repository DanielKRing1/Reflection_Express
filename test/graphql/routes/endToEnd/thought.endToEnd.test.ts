import http from "http";
import supertest from "supertest";

import server from "../../../../src/index";
import prisma from "../../../../src/prisma/client";
import { Inkling, Thought } from "@prisma/client";
import {
    advanceFakeTimer,
    useFakeTimer,
    useRealTimer,
} from "../../../../jest/utils/time";

const USER_ID1 = "1";
const USER_ID2 = "2";
const PASSWORD1 = "123";
const PASSWORD2 = "456";
const JOURNAL_NAME1 = "1";
const JOURNAL_NAME2 = "2";
const JOURNAL_NAME3 = "3";
const USER1_JOURNAL1_INKLINGS1 = [
    "User1 Inkling1 Journal1",
    "User1 Inkling2 Journal1",
    "User1 Inkling3 Journal1",
    "User1 Inkling4 Journal1",
    "User1 Inkling5 Journal1",
];
const USER1_JOURNAL1_INKLINGS2 = [
    "User1 Inkling6 Journal1",
    "User1 Inkling7 Journal1",
    "User1 Inkling8 Journal1",
    "User1 Inkling9 Journal1",
    "User1 Inkling10 Journal1",
];
const USER1_JOURNAL2_INKLINGS1 = [
    "User1 Inkling6 Journal2",
    "User1 Inkling7 Journal2",
    "User1 Inkling3 Journal2",
    "User1 Inkling9 Journal2",
    "User1 Inkling10 Journal2",
];
const USER2_JOURNAL3_INKLINGS1 = [
    "User2 Inkling1 Journal3",
    "User2 Inkling2 Journal3",
    "User2 Inkling3 Journal3",
    "User2 Inkling4 Journal3",
    "User2 Inkling5 Journal3",
];

describe("JournalEntry GraphQL server", () => {
    let s: http.Server;
    let agent: supertest.SuperAgentTest;

    let journal1Id: number;
    let journal2Id: number;
    let journal3Id: number;
    let inklings: Inkling[] = [];
    let journal1Thoughts: Thought[] = [];

    beforeAll(async () => {
        // Mock Date constructor
        useFakeTimer(new Date(2023, 4, 16));

        // Clear Database
        await prisma.$queryRaw`TRUNCATE TABLE "User" CASCADE;`;
        await prisma.$queryRaw`ALTER SEQUENCE "Journal_id_seq" RESTART WITH 1;`;

        // Create agent
        s = await server;
        agent = supertest.agent(s);

        // Create User/ Login
        await agent
            .post("/login/create-user")
            .send({ userId: USER_ID1, password: PASSWORD1 });
    });

    it("Should create User 1 Journal 1", async () => {
        const queryData = {
            query: `mutation CreateJournal($journalName: String!) {
                createJournal(journalName: $journalName) {
                    id
                    userId
                    name
                }
              }`,
            variables: {
                journalName: JOURNAL_NAME1,
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        journal1Id = JSON.parse(res.text).data.createJournal.id;
        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should create 5 User 1 Journal 1 Inklings", async () => {
        const queryData = {
            query: `mutation CommitInklings($commitInklingsJournalId: BigInt!, $inklingTexts: [String]!) {
                commitInklings(journalId: $commitInklingsJournalId, inklingTexts: $inklingTexts)
                {
                    timeId
                    journalId
                    text
                }
              }`,
            variables: {
                commitInklingsJournalId: JOURNAL_NAME1,
                inklingTexts: USER1_JOURNAL1_INKLINGS1,
            },
        };

        const res = await agent.post("/graphql").send(queryData);
        inklings = JSON.parse(res.text).data.commitInklings.sort(
            (a: any, b: any) => new Date(a).getTime() - new Date(b).getTime()
        );

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get convert User 1 Journal 1 Inkling into Thoughts", async () => {
        const queryData = {
            query: `mutation CreateJournalEntry($createJournalEntryJournalId: BigInt!, $keepIdsInkling: [DateTime]!, $keepIdsThought: [DateTime]!, $discardIdsThought: [DateTime]!, $discardIdsInkling: [DateTime]!) {
                createJournalEntry(journalId: $createJournalEntryJournalId, keepIdsInkling: $keepIdsInkling, keepIdsThought: $keepIdsThought, discardIdsThought: $discardIdsThought, discardIdsInkling: $discardIdsInkling)
              }`,
            variables: {
                createJournalEntryJournalId: journal1Id,
                keepIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 4),
                discardIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(4, 5),
                keepIdsThought: [],
                discardIdsThought: [],
            },
        };

        const res = await agent.post("/graphql").send(queryData);
    });

    it("Should get User 1 Journal 1 Thoughts 1 - 5", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal1Id,
                thoughtIds: inklings.map((inkling: Inkling) => inkling.timeId),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        journal1Thoughts = JSON.parse(res.text).data.thoughts;
        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get User 1 Journal 1 Thoughts only 1 - 2", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal1Id,
                thoughtIds: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 2),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should create 5 more User 1 Journal 1 Inklings", async () => {
        // Advance 10 seconds
        advanceFakeTimer(1000 * 10);

        const queryData = {
            query: `mutation CommitInklings($commitInklingsJournalId: BigInt!, $inklingTexts: [String]!) {
                commitInklings(journalId: $commitInklingsJournalId, inklingTexts: $inklingTexts)
                {
                    timeId
                    journalId
                    text
                }
              }`,
            variables: {
                commitInklingsJournalId: JOURNAL_NAME1,
                inklingTexts: USER1_JOURNAL1_INKLINGS2,
            },
        };

        const res = await agent.post("/graphql").send(queryData);
        inklings = JSON.parse(res.text).data.commitInklings.sort(
            (a: any, b: any) => new Date(a).getTime() - new Date(b).getTime()
        );

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get convert User 1 Journal 1 Inkling into Thoughts", async () => {
        const queryData = {
            query: `mutation CreateJournalEntry($createJournalEntryJournalId: BigInt!, $keepIdsInkling: [DateTime]!, $keepIdsThought: [DateTime]!, $discardIdsThought: [DateTime]!, $discardIdsInkling: [DateTime]!) {
                createJournalEntry(journalId: $createJournalEntryJournalId, keepIdsInkling: $keepIdsInkling, keepIdsThought: $keepIdsThought, discardIdsThought: $discardIdsThought, discardIdsInkling: $discardIdsInkling)
              }`,
            variables: {
                createJournalEntryJournalId: journal1Id,
                keepIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 4),
                discardIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(4, 5),
                keepIdsThought: [],
                discardIdsThought: [],
            },
        };

        const res = await agent.post("/graphql").send(queryData);
    });

    it("Should get User 1 Journal 1 Thoughts 5 - 10", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal1Id,
                thoughtIds: inklings.map((inkling: Inkling) => inkling.timeId),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get User 1 Journal 1 Thoughts only 4 - 5", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal1Id,
                thoughtIds: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(3),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should create User 1 Journal 2", async () => {
        const queryData = {
            query: `mutation CreateJournal($journalName: String!) {
                createJournal(journalName: $journalName) {
                    id
                    userId
                    name
                }
              }`,
            variables: {
                journalName: JOURNAL_NAME2,
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        journal2Id = JSON.parse(res.text).data.createJournal.id;
        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should create 5 User 1 Journal 2 Inklings", async () => {
        const queryData = {
            query: `mutation CommitInklings($commitInklingsJournalId: BigInt!, $inklingTexts: [String]!) {
                commitInklings(journalId: $commitInklingsJournalId, inklingTexts: $inklingTexts)
                {
                    timeId
                    journalId
                    text
                }
              }`,
            variables: {
                commitInklingsJournalId: JOURNAL_NAME2,
                inklingTexts: USER1_JOURNAL2_INKLINGS1,
            },
        };

        const res = await agent.post("/graphql").send(queryData);
        inklings = JSON.parse(res.text).data.commitInklings.sort(
            (a: any, b: any) => new Date(a).getTime() - new Date(b).getTime()
        );

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get convert User 1 Journal 2 Inkling into Thoughts", async () => {
        const queryData = {
            query: `mutation CreateJournalEntry($createJournalEntryJournalId: BigInt!, $keepIdsInkling: [DateTime]!, $keepIdsThought: [DateTime]!, $discardIdsThought: [DateTime]!, $discardIdsInkling: [DateTime]!) {
                createJournalEntry(journalId: $createJournalEntryJournalId, keepIdsInkling: $keepIdsInkling, keepIdsThought: $keepIdsThought, discardIdsThought: $discardIdsThought, discardIdsInkling: $discardIdsInkling)
              }`,
            variables: {
                createJournalEntryJournalId: journal2Id,
                keepIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 4),
                discardIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(4, 5),
                keepIdsThought: [],
                discardIdsThought: [],
            },
        };

        const res = await agent.post("/graphql").send(queryData);
    });

    it("Should get User 1 Journal 2 Thoughts 1 - 5", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal2Id,
                thoughtIds: inklings.map((inkling: Inkling) => inkling.timeId),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get User 1 Journal 2 Thoughts only 1 - 2", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal2Id,
                thoughtIds: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 2),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should log in as User 2", async () => {
        // "Logout"
        agent = supertest.agent(s);

        // Create new User/ Login
        await agent
            .post("/login/create-user")
            .send({ userId: USER_ID2, password: PASSWORD2 });
    });

    it("Should create User 2 Journal 3", async () => {
        const queryData = {
            query: `mutation CreateJournal($journalName: String!) {
                createJournal(journalName: $journalName) {
                    id
                    userId
                    name
                }
              }`,
            variables: {
                journalName: JOURNAL_NAME3,
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        journal3Id = JSON.parse(res.text).data.createJournal.id;
        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should create 5 User 2 Journal 3 Inklings", async () => {
        const queryData = {
            query: `mutation CommitInklings($commitInklingsJournalId: BigInt!, $inklingTexts: [String]!) {
                commitInklings(journalId: $commitInklingsJournalId, inklingTexts: $inklingTexts)
                {
                    timeId
                    journalId
                    text
                }
              }`,
            variables: {
                commitInklingsJournalId: JOURNAL_NAME3,
                inklingTexts: USER2_JOURNAL3_INKLINGS1,
            },
        };

        const res = await agent.post("/graphql").send(queryData);
        inklings = JSON.parse(res.text).data.commitInklings.sort(
            (a: any, b: any) => new Date(a).getTime() - new Date(b).getTime()
        );

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get convert User 2 Journal 3 Inkling into Thoughts", async () => {
        const queryData = {
            query: `mutation CreateJournalEntry($createJournalEntryJournalId: BigInt!, $keepIdsInkling: [DateTime]!, $keepIdsThought: [DateTime]!, $discardIdsThought: [DateTime]!, $discardIdsInkling: [DateTime]!) {
                createJournalEntry(journalId: $createJournalEntryJournalId, keepIdsInkling: $keepIdsInkling, keepIdsThought: $keepIdsThought, discardIdsThought: $discardIdsThought, discardIdsInkling: $discardIdsInkling)
              }`,
            variables: {
                createJournalEntryJournalId: journal3Id,
                keepIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 4),
                discardIdsInkling: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(4, 5),
                keepIdsThought: [],
                discardIdsThought: [],
            },
        };

        const res = await agent.post("/graphql").send(queryData);
    });

    it("Should get User 2 Journal 3 Thoughts 1 - 5", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal3Id,
                thoughtIds: inklings.map((inkling: Inkling) => inkling.timeId),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should get User 2 Journal 3 Thoughts only 1 - 2", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal3Id,
                thoughtIds: inklings
                    .map((inkling: Inkling) => inkling.timeId)
                    .slice(0, 2),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    it("Should fail to get User 1 Journal 1 Thoughts", async () => {
        const queryData = {
            query: `query Thoughts($journalId: ID!, $thoughtIds: [DateTime]!) {
                thoughts(journalId: $journalId, thoughtIds: $thoughtIds) {
                  timeId
                  journalId
                  text
                }
              }`,
            variables: {
                journalId: journal1Id,
                thoughtIds: journal1Thoughts.map(
                    (thought: Thought) => thought.timeId
                ),
            },
        };

        const res = await agent.post("/graphql").send(queryData);

        expect(JSON.parse(res.text)).toMatchSnapshot();
    });

    afterAll(async () => {
        // Unmock Date constructor
        useRealTimer();

        await prisma.$queryRaw`TRUNCATE TABLE "User" CASCADE;`;
        await prisma.$queryRaw`ALTER SEQUENCE "Journal_id_seq" RESTART WITH 1;`;

        await prisma.$disconnect();
        await s.close();
    });
});
