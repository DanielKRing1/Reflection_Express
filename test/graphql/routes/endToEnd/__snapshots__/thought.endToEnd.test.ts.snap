// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JournalEntry GraphQL server Should create 5 User 1 Journal 1 Inklings 1`] = `
{
  "data": {
    "commitInklings": [
      {
        "journalId": "1",
        "text": "User1 Inkling1 Journal1",
        "timeId": 1684220400000,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling2 Journal1",
        "timeId": 1684220400001,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling3 Journal1",
        "timeId": 1684220400002,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling4 Journal1",
        "timeId": 1684220400003,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling5 Journal1",
        "timeId": 1684220400004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should create 5 User 1 Journal 2 Inklings 1`] = `
{
  "data": {
    "commitInklings": [
      {
        "journalId": "2",
        "text": "User1 Inkling6 Journal2",
        "timeId": 1684220410000,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling7 Journal2",
        "timeId": 1684220410001,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling3 Journal2",
        "timeId": 1684220410002,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling9 Journal2",
        "timeId": 1684220410003,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling10 Journal2",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should create 5 User 2 Journal 3 Inklings 1`] = `
{
  "data": {
    "commitInklings": [
      {
        "journalId": "3",
        "text": "User2 Inkling1 Journal3",
        "timeId": 1684220410000,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling2 Journal3",
        "timeId": 1684220410001,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling3 Journal3",
        "timeId": 1684220410002,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling4 Journal3",
        "timeId": 1684220410003,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling5 Journal3",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should create 5 more User 1 Journal 1 Inklings 1`] = `
{
  "data": {
    "commitInklings": [
      {
        "journalId": "1",
        "text": "User1 Inkling6 Journal1",
        "timeId": 1684220410000,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling7 Journal1",
        "timeId": 1684220410001,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling8 Journal1",
        "timeId": 1684220410002,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling9 Journal1",
        "timeId": 1684220410003,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling10 Journal1",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should create User 1 Journal 1 1`] = `
{
  "data": {
    "createJournal": {
      "id": "1",
      "name": "1",
      "userId": "1",
    },
  },
}
`;

exports[`JournalEntry GraphQL server Should create User 1 Journal 2 1`] = `
{
  "data": {
    "createJournal": {
      "id": "2",
      "name": "2",
      "userId": "1",
    },
  },
}
`;

exports[`JournalEntry GraphQL server Should create User 1 Journal 3 1`] = `
{
  "data": {
    "createJournal": {
      "id": "3",
      "name": "3",
      "userId": "2",
    },
  },
}
`;

exports[`JournalEntry GraphQL server Should fail to get User 1 Journal 1 Thoughts 1`] = `
{
  "data": {
    "thoughts": [],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 1 Journal 1 Thoughts 1 - 5 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "1",
        "text": "User1 Inkling1 Journal1",
        "timeId": 1684220400000,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling2 Journal1",
        "timeId": 1684220400001,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling3 Journal1",
        "timeId": 1684220400002,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling4 Journal1",
        "timeId": 1684220400003,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling5 Journal1",
        "timeId": 1684220400004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 1 Journal 1 Thoughts 5 - 10 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "1",
        "text": "User1 Inkling6 Journal1",
        "timeId": 1684220410000,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling7 Journal1",
        "timeId": 1684220410001,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling8 Journal1",
        "timeId": 1684220410002,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling9 Journal1",
        "timeId": 1684220410003,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling10 Journal1",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 1 Journal 1 Thoughts only 1 - 2 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "1",
        "text": "User1 Inkling1 Journal1",
        "timeId": 1684220400000,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling2 Journal1",
        "timeId": 1684220400001,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 1 Journal 1 Thoughts only 4 - 5 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "1",
        "text": "User1 Inkling9 Journal1",
        "timeId": 1684220410003,
      },
      {
        "journalId": "1",
        "text": "User1 Inkling10 Journal1",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 1 Journal 2 Thoughts 1 - 5 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "2",
        "text": "User1 Inkling6 Journal2",
        "timeId": 1684220410000,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling7 Journal2",
        "timeId": 1684220410001,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling3 Journal2",
        "timeId": 1684220410002,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling9 Journal2",
        "timeId": 1684220410003,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling10 Journal2",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 1 Journal 2 Thoughts only 1 - 2 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "2",
        "text": "User1 Inkling6 Journal2",
        "timeId": 1684220410000,
      },
      {
        "journalId": "2",
        "text": "User1 Inkling7 Journal2",
        "timeId": 1684220410001,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 2 Journal 3 Thoughts 1 - 5 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "3",
        "text": "User2 Inkling1 Journal3",
        "timeId": 1684220410000,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling2 Journal3",
        "timeId": 1684220410001,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling3 Journal3",
        "timeId": 1684220410002,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling4 Journal3",
        "timeId": 1684220410003,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling5 Journal3",
        "timeId": 1684220410004,
      },
    ],
  },
}
`;

exports[`JournalEntry GraphQL server Should get User 2 Journal 3 Thoughts only 1 - 2 1`] = `
{
  "data": {
    "thoughts": [
      {
        "journalId": "3",
        "text": "User2 Inkling1 Journal3",
        "timeId": 1684220410000,
      },
      {
        "journalId": "3",
        "text": "User2 Inkling2 Journal3",
        "timeId": 1684220410001,
      },
    ],
  },
}
`;
